sympy/solvers/tests/test_solveset.py

```diff
@@ -1086,3 +1071,39 @@ def test_issue_8715():
         (Interval.open(-2, oo) - FiniteSet(0))
     assert solveset(eq.subs(x,log(x)), x, S.Reals) == \
         Interval.open(exp(-2), oo) - FiniteSet(1)
+
+def test_issue_9824():
+    assert solveset(sin(x)**2 - 2*sin(x) + 1, x, domain=S.Reals) ==\
+     ImageSet(Lambda(n, 2*n*pi + pi/2), S.Integers)
+
+
+def test_issue_9531_and_9606():
+    # complex solution
+    assert solveset(sinh(x)) == ImageSet(Lambda(n, n*I*pi), S.Integers)
+    # real solution
+    assert solveset(sinh(x), x, S.Reals) == FiniteSet(0)
+
+
+
+def test_issue_7914():
+    n1 = Dummy('n1')
+    n2 = Dummy('n2')
+    assert solveset(sin(2*x)*cos(x) + cos(2*x)*sin(x) -1, x) ==\
+    ImageSet(Lambda(n1, pi*(4*n1 - 3)/6), S.Integers) + ImageSet(Lambda(n2, 2*n2*pi - pi/2), S.Integers)
+
+
+def test_issue_10671():
+    assert solveset(sin(y), y, Interval(0, pi)) == \
+    Intersection(Interval(0, pi), ImageSet(Lambda(n, n*pi), S.Integers))
+
+
```

Comment:
If you try these examples in current master branch we don't get these simplified solution.